atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/HostLocationSupplier.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.pool;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public final class HostLocationSupplier implements Supplier<Optional<HostLocation>> {
+
+    private String snitch;
```

Comment:
In general, you should try to inject things rather than using setters. The advantage is that your objects are always in a fully constructed state, rather than there being a middle phase where it somehow has to be initialized. In this, case, I'd be tempted to construct this object with a supplier of snitch, rather than having a field which causes the behaviour to switch out.